libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -161,6 +161,40 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
         )
     }
 
+    _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_INLINE_VISIBILITY
+
+    arrival_token arrive_tx(_CUDA_VSTD::ptrdiff_t arrive_count_update,
+                            _CUDA_VSTD::ptrdiff_t transaction_count_update) {
```

Comment:
We need to __uglify our code to guard against macro pollution
```suggestion
    arrival_token arrive_tx(_CUDA_VSTD::ptrdiff_t __arrive_count_update,
                            _CUDA_VSTD::ptrdiff_t __transaction_count_update) {
```

Reply:
This applies throughout

Reply:
Thanks for the pointer. I have added double underscores throughout.